.github/workflows/cd.yaml

```diff
@@ -0,0 +1,54 @@
+
+name: CD Pipeline
+
+on:
+  push:
+    branches: [ feature/continuous-delivery ]
+  workflow_dispatch:
+
+jobs:
+  publish-to-pypi:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the repository with full history depth to retrieve all tags
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools wheel
+          python -m pip install --upgrade twine
+          python -m pip install --upgrade setuptools_scm
+          python -m pip install --upgrade build
+          python -m pip install --upgrade semver
+
+      - name: Increment the git tag
+        run: |
+          # get the current git tag
+          git fetch --tags origin
+          CURRENT_TAG=$(git describe --tags --abbrev=0)
+          # Increment the tag by a minor version with the semver python package
+          NEW_TAG=$(python -c "import semver; print(semver.bump_minor('$CURRENT_TAG'))")
```

Comment:
So this will always bump a minor version. What would be the workflow to bump a major? We would do it manually if needed?

Reply:
Good question. We could have an environment variable MAJOR_UPDATE which indicates if the bumb is major or not (it would be false by default). I can try to implement this if needed.